test/integration/pack/test_pack.py

```diff
@@ -990,3 +990,279 @@ def test_verbosity(cartridge_cmd, project_without_dependencies, pack_format):
+            "dependency01 (<< 3)",
+            "dependency02 (= 2.5)",
+            "dependency03"
+        )
+
+        assert_dependencies_deb(find_archive(tmpdir, project.name, 'deb'), deps, tmpdir)
+    else:
+        deps = (
+            ("dependency01", 0x08 | 0x04, "1.2"),  # >=
+            ("dependency01", 0x02, "3"),  # <
+            ("dependency02", 0x08, "2.5"),  # =
+            ("dependency03", 0, "")
+        )
+
+        assert_dependencies_rpm(find_archive(tmpdir, project.name, 'rpm'), deps)
+
+
+@pytest.mark.parametrize('pack_format', ['docker', 'tgz'])
+def test_dependencies_not_rpm_deb(cartridge_cmd, project_without_dependencies, pack_format, tmpdir):
+    project = project_without_dependencies
+
+    cmd = [
+        cartridge_cmd,
+        "pack", pack_format,
+        "--deps-file", "dependencies.txt",
+        project.path,
+    ]
+
+    warning_message = "You specified the --deps-file flag, but you are not packaging RPM or DEB. Flag will be ignored"
+    rc, output = run_command_and_get_output(cmd, cwd=tmpdir)
+    assert rc == 0
+    assert warning_message in output
+
+    cmd = [
+        cartridge_cmd,
+        "pack", pack_format,
+        "--deps", "dependencies01",
+        project.path
+    ]
+
+    warning_message = "You specified the --deps flag, but you are not packaging RPM or DEB. Flag will be ignored"
+    rc, output = run_command_and_get_output(cmd, cwd=tmpdir)
+    assert rc == 0
+    assert warning_message in output
+
+
+@pytest.mark.parametrize('pack_format', ['deb', 'rpm'])
+def test_dependencies_same_flags(cartridge_cmd, project_without_dependencies, pack_format, tmpdir):
+    project = project_without_dependencies
+
+    deps_filepath = os.path.join(tmpdir, "deps.txt")
+    with open(deps_filepath, "w") as f:
+        f.write("")
+
+    cmd = [
+        cartridge_cmd,
+        "pack", pack_format,
+        "--deps-file", deps_filepath,
+        "--deps", "dependency01",
+        project.path,
+    ]
+
+    if platform.system() == 'Darwin':
+        cmd.append('--use-docker')
+
+    rc, output = run_command_and_get_output(cmd, cwd=tmpdir)
+    assert rc == 1
+    assert "You can't specify --deps and --deps-file flags at the same time" in output
+
+
+@pytest.mark.parametrize('pack_format', ['deb', 'rpm'])
+def test_dependencies_file_not_exist(cartridge_cmd, project_without_dependencies, pack_format, tmpdir):
+    project = project_without_dependencies
+
+    cmd = [
+        cartridge_cmd,
+        "pack", pack_format,
+        "--deps-file", "not_exist_file.txt",
+        project.path,
+    ]
+
+    if platform.system() == 'Darwin':
+        cmd.append('--use-docker')
+
+    rc, output = run_command_and_get_output(cmd, cwd=tmpdir)
+    assert rc == 1
+    assert "Invalid path to file with dependencies" in output
+
+
+@pytest.mark.parametrize('pack_format', ['deb', 'rpm'])
+def test_broken_dependencies(cartridge_cmd, project_without_dependencies, pack_format, tmpdir):
```

Comment:
Seems redundantly - you've tested that in unit tests.

Reply:
No, unit tests do not check for incorrect dependencies (only correct)

Reply:
It would be better to move such tests into units.
Unit tests check that files are parsed correctly, they can test "error" cases too.
Integration tests check the whole CLI behavior, so you can place here one "error" case and validate that user will see nice error.